core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -56,8 +51,12 @@ public SetupWizard(Jenkins j) throws IOException {
             HudsonPrivateSecurityRealm securityRealm = new HudsonPrivateSecurityRealm(false, false, null);
             j.setSecurityRealm(securityRealm);
             String randomUUID = UUID.randomUUID().toString().replace("-", "").toLowerCase(Locale.ENGLISH);
-            admin = securityRealm.createAccount(SetupWizard.initialSetupAdminUserName, randomUUID);
-            admin.addProperty(new SetupWizard.AuthenticationKey(randomUUID));
+            
+            // create an admin user
+            securityRealm.createAccount(SetupWizard.initialSetupAdminUserName, randomUUID);
+            
+            // JENKINS-33599 - write to a file in the jenkins home directory
+            FileUtils.write(getInitialAdminPasswordFile(), randomUUID);
```

Comment:
@oleg-nenashev yes, startup of the installer should fail if it can't read/write the file, that's the only way a user could really continue